### PR TITLE
Rename check_examples.py to test_check_examples.py and add existing example JSON files

### PR DESCRIPTION
**Pull request by njzjz**
It seems that `check_examples.py` did not run in the CI (?) since the filename doesn't start with `test`.

The Python unittest module has a different behavior from pytest. Pytest runs all tests regardless of the filename.

## Summary by CodeRabbit

- **Tests**
  - Added new test input files for `calypso`, `water`, and `ch4` examples in the test suite.

### CODE REVIEW

**Comment by zjgemi** [comment]
Is it necessary to add existing example JSON files to this test in this PR?

**Comment by njzjz** [comment]
> Is it necessary to add existing example JSON files to this test in this PR?

Which one is new?

**Comment by zjgemi** [comment]
> > Is it necessary to add existing example JSON files to this test in this PR?
> 
> Which one is new?

calyspo/input.test.json
water/input_distill.json
water/input_dpgen.json
water/input_multitask.json

File: tests/test_check_examples.py [hunk 1]
```diff
@@ -14,6 +14,10 @@
     p_examples / "almg" / "input.json",
     # p_examples / "almg" / "input-v005.json",
     # p_examples / "almg" / "dp_template.json",
+    p_examples / "calyspo" / "input.test.json",
```
Comment (njzjz) [inline tests/test_check_examples.py:17]:
```suggestion
    p_examples / "calypso" / "input.test.json",
```

**Review by zjgemi — APPROVED** [review]

**Pull request merged**